src/backend/utils/compatibility_layers.ts

```diff
@@ -562,6 +563,18 @@ export async function getWineFlagsArray(
 export const getUmuPath = async () =>
   searchForExecutableOnPath('umu-run').then((path) => path ?? defaultUmuPath)
 
+export async function usingNixUmu(): Promise<boolean> {
+  return searchForExecutableOnPath('umu-run').then((path) => {
+    if (path) {
+      return false
+    } else {
+      return realpath(path as string).then((path) =>
+        path.startsWith('/nix/store/')
+      )
+    }
+  })
+}
```

Comment:
This function doesn't look right
- If a path is found, you're always returning `false`
- If a path isn't found, you treat the returned `null` as a string and try to call `realpath` on it?

Perhaps there's a missing `!` in the condition?

In general I'd use `await` here to make this more readable, along with the existing `getUmuPath`:
```ts
  const umuPath = await getUmuPath()
  const realUmuPath = await realpath(umuPath)
  return realUmuPath.startsWith('/nix/store/')
```

I'm also not sure whether `realpath` is necessary here, but I assume you've confirmed that